solutions/silver/cf-51C.mdx

```diff
@@ -0,0 +1,91 @@
+---
+id: cf-51C
+source: CF
+title: Three Base Stations
+author: KJ Karaisz
+---
+
+<Spoiler title = "Hint 1">
+
+What if you were given $d$, and told to compute whether or not it is possible to cover all houses?
+
+</Spoiler>
+
+<Spoiler title = "Hint 2">
+
+Look at just one house. A station has to go somewhere within range of it. Where's the best place to put it?
+
+</Spoiler>
+
+<Spoiler title = "Solution">
+
+## Explanation
+
+Let's binary search on the answer.
+
+Consider the leftmost house. We have to place a station within range of this
+house. Because there are no houses to the left, and therefore no reason for the
+station's range to extend to the left, an optimal place to put it is as far as
+possible to the right, at $\texttt{pos}+d$.
+
+When our first station's range runs out, we can use similar logic to place the
+next two—there's no need for the range to extend to the left of the leftmost
+uncovered house, so we can place the next station as far as possible to the
+right again.
+
+Our test function returns true if all the houses can be covered placing the
+stations in this manner.
+
+## Implementation
+
+**Time Complexity:** $\mathcal{O}(N\log N)$
+
+<LanguageSection>
+
+<CPPSection>
+
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+int main() {
+	int n;
+	cin >> n;
+	vector<int> x(n);
+	for (int i = 0; i < n; i++) { cin >> x[i]; }
+	ranges::sort(x);
```

Comment:
most of the code in the guide uses `std::sort` instead of `range::sort`

Reply:
I think `ranges` is a wonderful and underused C++20 feature. But fine.